src/input.rs

```diff
@@ -134,6 +136,33 @@ impl<'a> Input<'a> {
         Ok(())
     }
 
+    fn handle_ctrl_u(&mut self) -> io::Result<()> {
+        self.input.replace_range(..self.cursor, "");
+        self.cursor = 0;
+        Ok(())
+    }
+
+    fn handle_ctrl_w(&mut self) -> io::Result<()> {
```

Comment:
not as sophisticated as the real thing as it only detects words separated by spaces for now